### PR TITLE
[SPARK-33434][PYTHON][DOCS] Added RuntimeConfig to PySpark docs 

### PR DESCRIPTION
**Pull request by Eric-Lemmon**
### What changes were proposed in this pull request?
Documentation for `SparkSession.conf.isModifiable` is missing from the Python API site, so we added a Configuration section to the Spark SQL page to expose docs for the `RuntimeConfig` class (the class containing `isModifiable`). Then a `:class:` reference to `RuntimeConfig` was added to the `SparkSession.conf` docstring to create a link there as well. 

### Why are the changes needed?
No docs were generated for `pyspark.sql.conf.RuntimeConfig`.

### Does this PR introduce _any_ user-facing change?
Yes--a new Configuration section to the Spark SQL page and a `Returns` section of the `SparkSession.conf` docstring, so this will now show a link to the `pyspark.sql.conf.RuntimeConfig` page. This is a change compared to both the released Spark version and the unreleased master branch.

### How was this patch tested?
First built the Python docs:
```bash
cd $SPARK_HOME/docs
SKIP_SCALADOC=1 SKIP_RDOC=1 SKIP_SQLDOC=1 jekyll serve
```
Then verified all pages and links:
1. Configuration link displayed on the API Reference page, and it clicks through to Spark SQL page:
http://localhost:4000/api/python/reference/index.html
image

2. Configuration section displayed on the Spark SQL page, and the RuntimeConfig link clicks through to the RuntimeConfig page:
http://localhost:4000/api/python/reference/pyspark.sql.html#configuration
image**

3. RuntimeConfig page displayed:
http://localhost:4000/api/python/reference/api/pyspark.sql.conf.RuntimeConfig.html
image

4. SparkSession.conf page displays the RuntimeConfig link, and it navigates to the RuntimeConfig page:
http://localhost:4000/api/python/reference/api/pyspark.sql.SparkSession.conf.html
image

### CODE REVIEW

**Comment by maropu** [comment]
ok to test

**Comment by maropu** [comment]
For reviews, could you put the screenshot of the updated doc in the PR description?

File: python/docs/source/reference/pyspark.sql.rst [hunk 1]
```diff
@@ -72,6 +72,11 @@ See also :class:`SparkSession`.
     SparkSession.udf
     SparkSession.version
 
+.. toctree::
+    :hidden:
+
+    pyspark.sql.conf.RuntimeConfig
```
Comment (HyukjinKwon) [inline python/docs/source/reference/pyspark.sql.rst:78]: Shall we just add another section "Configuration"? e.g.)

```
Configuration
-------------

.. currentmodule:: pyspark.sql.conf

.. autosummary::
    :toctree: api/

    RuntimeConfig
```

Comment (Eric-Lemmon) [reply python/docs/source/reference/pyspark.sql.rst:78]: Yes, Hyukjin, that will work. I was unsure if that should be elevated to its own section, but it is definitely a simpler / more elegant solution. Thank you for the suggestion.  I will update the PR today.

Comment (Eric-Lemmon) [reply python/docs/source/reference/pyspark.sql.rst]: I pushed up changes with your suggestion, removed the RuntimeConfig rst file to allow autosummary to generate it, and updated the PR description.  Thank you for the review, Hyukjin.

**Review by srowen — APPROVED** [review]

**Review by HyukjinKwon — COMMENTED** [review]
lgtm2

**Comment by HyukjinKwon** [comment]
It looks merged into master only. I merged it into branch-3.1 too.

**Pull request closed without merging**